src/classes/GS_AdminSetup.cls

```diff
@@ -0,0 +1,293 @@
+        ci.image = item.Image__c;
+        ci.pos = item.Position__c.intValue();
+        if (item.Has_Link__c) {
+            ci.link = new ChecklistItemLink(item.Link_Label__c, item.Link_URL__c);
+        }
+        if (item.Primary_Button_Type__c != DISABLED) {
+            ci.principalBtn = new ChecklistItemButton(item.Primary_Button_Type__c, item.Primary_Button_Label__c, item.Primary_Button_Value__c);
+        }
+        if (item.Secondary_Button_Type__c != DISABLED) {
+            ci.secondaryBtn = new ChecklistItemButton(item.Secondary_Button_Type__c, item.Secondary_Button_Label__c, item.Secondary_Button_Value__c);
+        }
+        return ci;
+    }
+
+    /*
+    * @description This Class is to wrap GS_Admin_Checklist_Section__mdt information
+    */
+    public class ChecklistSection {
+        /*
+        * @description List of items in the secction
+        * @type List<ChecklistItem>
+        */
+        @AuraEnabled
+        public List<ChecklistItem> items = new List<ChecklistItem>();
+        /*
+        * @description Name of title label
+        * @type String
+        */
+        @AuraEnabled
+        public String title;
+        /*
+        * @description Name of description label
+        * @type String
+        */
+        @AuraEnabled
+        public String description;
+        /*
+        * @description Section position
+        * @type
+        */
+        @AuraEnabled
+        public Integer pos;
+        /*
+        * @description Metadata Id
+        * @type Id
+        */
+        @AuraEnabled
+        public Id id;
+    }
+
+    /*
+    * @description This Class is to wrap GS_Admin_Checklist_Item__mdt information
+    */
+    public class ChecklistItem implements Comparable {
+        /**
+        * @description Metadata id
+        * @type Id
+        */
+        @AuraEnabled
+        public Id id;
+        /**
+        * @description Item title label Name
+        * @type String
+        */
+        @AuraEnabled
+        public String title;
+        /*
+        * @description  Item position
+        * @type Integer
+        */
+        @AuraEnabled
+        public Integer pos;
+        /**
+        * @description Item describtion label Name
+        * @type String
+        */
+        @AuraEnabled
+        public String description;
+        /**
+        * @description Item image name
+        * @type String
+        */
+        @AuraEnabled
+        public String image;
+        /**
+        * @description Item extra info label name
+        * @type String
+        */
+        @AuraEnabled
+        public String extraInfo;
+        /**
+        * @description Item link info
+        * @type ChecklistItemLink
+        */
+        @AuraEnabled
+        public ChecklistItemLink link;
+        /**
+        * @description Item principal button info
+        * @type ChecklistItemLink
+        */
+        @AuraEnabled
+        public ChecklistItemButton principalBtn;
```

Comment:
Variable, method and property names should also be renamed from "principal" to "primary".